### PR TITLE
update: a11y stability changes

### PR DESCRIPTION
**Pull request by 07souravkunda**
Adding stability v2 changes for accessibility.

- Fix accessiblity run for versions array. (versions: ['latest', 'latest-1'])
- Update accessibility build start api to v2, to get the updated extension version.
- Update scripts for accessibility scan, save results, get results and get results summary.
- Add BROWSERSTACK_ACCESSIBLITY_DEBUG environment variable to have debug logs.
- Use cy.wrap to chain the accessibility promises.
- Add warning message for headless and chrome version. < 94 for accessibility.

### CODE REVIEW

File: bin/accessibility-automation/cypress/index.js [hunk 1]
```diff
@@ -1,115 +1,356 @@
...
-        window.parent.dispatchEvent(e);
-        if (dataForExtension.saveResults !== true )
-          resolve();
-      });
-    }
+        const payloadToSend = {
+          "saveResults": shouldScanTestForAccessibility,
+          "testDetails": {
+            "name": attributes.title,
+            "testRunId": '5058', // variable not consumed, shouldn't matter what we send
```
Comment (kamal-kaur04) [inline bin/accessibility-automation/cypress/index.js:315]: This isn't returned in the build start?

File: bin/accessibility-automation/cypress/index.js [hunk 2]
```diff
@@ -1,115 +1,356 @@
...
 
-  } catch {}
-});
+      } catch (er) {
+      }
+    })
+  });
+})
+
+Cypress.Commands.add('perfromScan', () => {
```
Comment (kamal-kaur04) [inline bin/accessibility-automation/cypress/index.js:337]:
```suggestion
Cypress.Commands.add('performScan', () => {
```
maybe?

**Review by kamal-kaur04 — CHANGES_REQUESTED** [review]

**Review by kamal-kaur04 — DISMISSED** [review]

File: bin/accessibility-automation/cypress/index.js [hunk 3]
```diff
@@ -1,140 +1,383 @@
...
+    browserStackLog('Performing scan form command ' + command.attributes.name);
+    cy.wrap(performScan(win, {method: command.attributes.name}), {timeout: 30000});
+  })
+})
+
+afterEach(() => {
+  const attributes = Cypress.mocha.getRunner().suite.ctx.currentTest;
+  cy.window().then(async (win) => {
+    let shouldScanTestForAccessibility = shouldScanForAccessibility(attributes);
+    if (!shouldScanTestForAccessibility) return cy.wrap({});
```
Comment (07souravkunda) [inline bin/accessibility-automation/cypress/index.js:289]: check if shouldScanTestForAccessibility can be replaced

File: bin/helpers/capabilityHelper.js [hunk 4]
```diff
@@ -147,14 +149,27 @@ const caps = (bsConfig, zip) => {
...
+    rootLevelAccessibility = bsConfig.run_settings.accessibility.toString() === 'true';
   }
   browserList.forEach((browserDetails, idx) => {
-    accessibilityPlatforms[idx] = (browserDetails.accessibility === undefined) ? rootLevelAccessibility : browserDetails.accessibility
+    accessibilityPlatforms[idx] = (browserDetails.accessibility === undefined) ? rootLevelAccessibility : browserDetails.accessibility;
+    if (browserDetails.browser && browserDetails.browser.toLowerCase() !== 'chrome') {
+      logger.warn(`Accessibility Automation will run only on Chrome browsers for ${browserDetails.platform}.`);
+    } else if (browserDetails.version && (!browserDetails.version.includes('latest') || browserDetails.version < 94)) {
+      logger.warn(`Accessibility Automation will run only on Chrome browser version greater than 94 for ${browserDetails.platform}.`);
+    }
```
Comment (07souravkunda) [inline bin/helpers/capabilityHelper.js:172]: add validation for headed

File: bin/helpers/capabilityHelper.js [hunk 5]
```diff
@@ -147,14 +149,27 @@ const caps = (bsConfig, zip) => {
...
   if (!Utils.isUndefined(bsConfig.run_settings.accessibility)) {
-    rootLevelAccessibility = bsConfig.run_settings.accessibility.toString() === 'true'
+    rootLevelAccessibility = bsConfig.run_settings.accessibility.toString() === 'true';
   }
   browserList.forEach((browserDetails, idx) => {
-    accessibilityPlatforms[idx] = (browserDetails.accessibility === undefined) ? rootLevelAccessibility : browserDetails.accessibility
+    accessibilityPlatforms[idx] = (browserDetails.accessibility === undefined) ? rootLevelAccessibility : browserDetails.accessibility;
+    if (browserDetails.browser && browserDetails.browser.toLowerCase() !== 'chrome') {
+      logger.warn(`Accessibility Automation will run only on Chrome browsers for ${browserDetails.platform}.`);
+    } else if (browserDetails.version && (!browserDetails.version.includes('latest') || browserDetails.version < 94)) {
```
Comment (Mirudhula-Nadar) [inline bin/helpers/capabilityHelper.js:170]: @07souravkunda `!browserDetails.version.includes('latest')` it should not have not condition currently its throwing error when we provide any other version like 118, 120

**Review by amaanbs — APPROVED** [review]

**Review by asambstack — APPROVED** [review]

**Pull request merged**